qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py

```diff
@@ -282,3 +300,165 @@ def clear_buffer_of_error_message(self) -> None:
+                             channels: Optional[constants.ChannelList] = None
+                             ) -> None:
+        """
+        This method specifies the measurement mode and the channels used
+        for measurements. This method must be entered to specify the
+        measurement mode. For the high speed spot measurements,
+        do not use this method.
+
+        Args:
+            mode: Measurement mode. See `constants.MM.Mode` for all possible
+                modes
+            channels: Measurement channel number. See `constants.ChannelList`
+                for all possible channels.
+        """
+        msg = MessageBuilder().mm(mode=mode, channels=channels).message
+        self.write(msg)
+
+    def get_measurement_mode(self) -> Dict[str, Union[constants.MM.Mode,
+                                                      List]]:
+        """
+        This method gets the measurement mode(MM) and the channels used
+        for measurements. It outputs a dictionary with 'mode' and
+        'channels' as keys.
+        """
+        msg = MessageBuilder().lrn_query(type_id=constants.LRN.
+                                         Type.TM_AV_CM_FMT_MM_SETTINGS)
+        response = self.ask(msg.message)
+        match = re.search('MM(?P<mode>.*?),(?P<channels>.*?)(;|$)', response)
+
+        if not match:
+            raise ValueError('Measurement Mode (MM) not found.')
+
+        out_dict: Dict[str, Union[constants.MM.Mode, List]] = {}
+        resp_dict = match.groupdict()
+        out_dict['mode'] = constants.MM.Mode(int(resp_dict['mode']))
+        out_dict['channels'] = list(map(int, resp_dict['channels'].split(',')))
+        return out_dict
+
+    def get_response_format_and_mode(self) -> \
+            Dict[str, Union[constants.FMT.Format, constants.FMT.Mode]]:
+        """
+        This method queries the the data output format and mode.
+        """
+        msg = MessageBuilder().lrn_query(type_id=constants.LRN.
+                                         Type.TM_AV_CM_FMT_MM_SETTINGS)
+        response = self.ask(msg.message)
+        match = re.search('FMT(?P<format>.*?),(?P<mode>.*?)(;|$)',
+                          response)
+
+        if not match:
+            raise ValueError('Measurement Mode (FMT) not found.')
+
+        out_dict: Dict[str, Union[constants.FMT.Format, constants.FMT.Mode]] \
+            = {}
+        resp_dict = match.groupdict()
+        out_dict['format'] = constants.FMT.Format(int(resp_dict[
+                                                          'format']))
+        out_dict['mode'] = constants.FMT.Mode(int(resp_dict['mode']))
+        return out_dict
+
+
+class IVSweepMeasurement(MultiParameter):
+    """
+    IV sweep measurement outputs a list of primary and secondary
+    parameter.
+
+    Args:
+        name: Name of the Parameter.
+        instrument: Instrument to which this parameter communicates to.
+    """
+
+    def __init__(self, name, instrument, **kwargs):
```

Comment:
Suggest adding types for name and instrument here

Reply:
done